central/graphql/resolvers/namespaces.go

```diff
@@ -135,8 +135,25 @@ func (resolver *Resolver) Namespaces(ctx context.Context, args PaginatedQuery) (
 		return nil, err
 	}
 
-	ns, err := namespace.ResolveMetadataOnlyByQuery(ctx, query, resolver.NamespaceDataStore)
-	return resolver.wrapNamespacesWithContext(ctx, ns, err)
+	var namespaces []*v1.Namespace
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		nsLoader, err := loaders.GetNamespaceLoader(ctx)
+		if err != nil {
+			return nil, err
+		}
+		metadataSlice, err := nsLoader.FromQuery(ctx, query)
+		if err != nil {
+			return nil, err
+		}
+		namespaces = make([]*v1.Namespace, 0, len(metadataSlice))
+		for _, metadata := range metadataSlice {
+			namespaces = append(namespaces, &v1.Namespace{Metadata: metadata})
+		}
+		return resolver.wrapNamespacesWithContext(ctx, namespaces, nil)
+	}
+
+	namespaces, err = namespace.ResolveMetadataOnlyByQuery(ctx, query, resolver.NamespaceDataStore)
```

Comment:
Had added loader usage to `namespace.ResolveMetadataOnlyByQuery` but moved it out because other functions in the file `central/namespaces/resolvers.go` are used by namespace service, which does not have initialize loaders. 

I would instead remove `ResolveMetadataOnlyByQuery` from  and just add its code to `Namespaces` graphQL resolver which is the only reference to `ResolveMetadataOnlyByQuery`.